.github/workflows/kubespray.yml

```diff
@@ -0,0 +1,290 @@
+        # See https://github.com/aws/aws-cli/issues/5262
+        AWS_EC2_METADATA_DISABLED: true
+      with:
+        args: route53 change-resource-record-sets --hosted-zone-id Z2STJRQSJO5PZ0 --change-batch file://apps/pipeline/config/exoscale/dns-${{ matrix.cluster }}.json
+
+    - name: Install mitogen to speedup kubespray
+      uses: docker://quay.io/kubespray/kubespray:v2.15.0
+      with:
+        # Note: It is important to work from the kubespray folder since this
+        # playbook will modify the `ansible.cfg`.
+        args: /bin/bash -c "cd kubespray && ansible-playbook mitogen.yml"
+
+    - name: Run kubespray
+      uses: docker://quay.io/kubespray/kubespray:v2.15.0
+      env:
+        ANSIBLE_CONFIG: ./kubespray/ansible.cfg
+      with:
+        args: /bin/bash -c "cd kubespray && ansible-playbook -i ../$CK8S_CONFIG_PATH/pipeline-${{ matrix.cluster }}-config/inventory.ini --private-key ../$CK8S_CONFIG_PATH/id_rsa --become --become-user=root cluster.yml"
+
+    - name: Import PGP key and configure GPG agent
+      run: ./apps/pipeline/setup-pgp.bash
+      env:
+        PGP_KEY: ${{ secrets.PGP_KEY }}
+        PGP_PASSPHRASE: ${{ secrets.PGP_PASSPHRASE }}
+
+    - name: Prepare for apps
+      env:
+        S3_ACCESS_KEY: ${{ secrets.EXOSCALE_API_KEY }}
+        S3_SECRET_KEY: ${{ secrets.EXOSCALE_API_SECRET }}
+      run: |
+        # Set public LB IP in the kubeconfig
+        control_plane_lb_ip_address="$(jq -r .outputs.control_plane_lb_ip_address.value $CK8S_CONFIG_PATH/pipeline-${{ matrix.cluster }}-config/terraform.tfstate)"
+        yq write --inplace "$CK8S_CONFIG_PATH/pipeline-${{ matrix.cluster }}-config/artifacts/admin.conf" clusters[0].cluster.server https://${control_plane_lb_ip_address}:6443
+        # Copy to correct location and encrypt
+        mkdir "$CK8S_CONFIG_PATH/.state"
+        cp "$CK8S_CONFIG_PATH/pipeline-${{ matrix.cluster }}-config/artifacts/admin.conf" "$CK8S_CONFIG_PATH/.state/kube_config_${{ matrix.cluster }}.yaml"
+        sops --config "$CK8S_CONFIG_PATH/.sops.yaml" -e -i "$CK8S_CONFIG_PATH/.state/kube_config_${{ matrix.cluster }}.yaml"
+        # Encrypt and set necessary secrets
+        sops --config "$CK8S_CONFIG_PATH/.sops.yaml" -e -i "$CK8S_CONFIG_PATH/secrets.yaml"
+        sops --config "$CK8S_CONFIG_PATH/.sops.yaml" --set "[\"objectStorage\"] {\"s3\": {\"accessKey\": \"${S3_ACCESS_KEY}\", \"secretKey\": \"${S3_SECRET_KEY}\"}}" "$CK8S_CONFIG_PATH/secrets.yaml"
+
+    - name: Install rook
+      env:
+        KUBECONFIG: /__w/compliantkubernetes-apps/compliantkubernetes-apps/apps/pipeline/config/exoscale/pipeline-${{ matrix.cluster }}-config/artifacts/admin.conf
+      run: |
+        ./deploy-rook.sh
+      working-directory: ./compliantkubernetes-kubespray/rook
+
+    # TODO: Some kind of test of Rook before we waste time waiting for apps to fail if it is broken...
+    # We may also want to move Rook to the apps bootstrap instead.
+
```

Comment:
I think we could create some test with `kubectl get cephclusters.ceph.rook.io -n rook-ceph rook-ceph -o wide`, but change so we wait for plase=ready or something similar.
Might not be necessary to do before this PR is merged, but if you want you could do some quick tests.

Reply:
I'll leave that for https://github.com/elastisys/ck8s-pipelines/issues/47